pkg/api/serialization_test.go

```diff
@@ -58,6 +61,15 @@ func fuzzInternalObject(t *testing.T, forVersion unversioned.GroupVersion, item
 	return item
 }
 
+func dataAsString(data []byte) string {
+	dataString := string(data)
+	if !strings.HasPrefix(dataString, "{") {
+		dataString = "\n" + hex.Dump(data)
+		proto.NewBuffer(make([]byte, 0, 1024)).DebugPrint("decoded object", data)
```

Comment:
It's a helper that shows using proto rules the structs and their types.
It's moderately useful when debugging proto here to have the machine say
"this is what I got" and compare that to what turned up in the struct (in
case there is a mismatch between the correct proto and an incorrect
deserializer)

On Mon, Apr 11, 2016 at 3:22 AM, Wojciech Tyczynski <
notifications@github.com> wrote:

> In pkg/api/serialization_test.go
> <https://github.com/kubernetes/kubernetes/pull/23806#discussion_r59162670>
> :
>
> > @@ -58,6 +61,15 @@ func fuzzInternalObject(t *testing.T, forVersion unversioned.GroupVersion, item
> >  	return item
> >  }
> >
> > +func dataAsString(data []byte) string {
> > +	dataString := string(data)
> > +	if !strings.HasPrefix(dataString, "{") {
> > +		dataString = "\n" + hex.Dump(data)
> > +		proto.NewBuffer(make([]byte, 0, 1024)).DebugPrint("decoded object", data)
>
> Just curious - what's that?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/23806/files/e6879a33f46577ce93328230a28453244c3a408f#r59162670>
>